src/types/methods.ts

```diff
@@ -0,0 +1,706 @@
+  max_redemptions?: number;
+  /**
+   * Is discount applied only to certain variants
+   */
+  is_limited_to_products?: boolean;
+}
+
+export interface DeleteDiscountOptions {
+  /**
+   * The ID of the discount to delete
+   */
+  id: number;
+}
+
+export interface GetDiscountRedemptionsOptions extends PaginatedOptions {
+  /**
+   * Filter discount redemptions by discount
+   */
+  discountId?: number;
+  /**
+   * Filter discount redemptions by order
+   */
+  orderId?: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"discount" | "order">;
+}
+
+export interface GetDiscountRedemptionOptions {
+  /**
+   * ID of the discount to retrieve
+   */
+  id: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"discount" | "order">;
+}
+
+export interface GetLicenseKeysOptions extends PaginatedOptions {
+  /**
+   * Filter license keys by store
+   */
+  storeId?: number;
+  /**
+   * Filter license keys by order
+   */
+  orderId?: number;
+  /**
+   * Filter license keys by order item
+   */
+  orderItemId?: number;
+  /**
+   * Filter license keys by product
+   */
+  productId?: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<
+    | "store"
+    | "customer"
+    | "order"
+    | "order-item"
+    | "product"
+    | "license-key-instances"
+  >;
+}
+
+export interface GetLicenseKeyOptions {
+  /**
+   * ID of the license key to retrieve
+   */
+  id: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<
+    | "store"
+    | "customer"
+    | "order"
+    | "order-item"
+    | "product"
+    | "license-key-instances"
+  >;
+}
+
+export interface GetLicenseKeyInstancesOptions extends PaginatedOptions {
+  /**
+   * Filter license key instances by license key
+   */
+  licenseKeyId?: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"license-key">;
+}
+
+export interface GetLicenseKeyInstanceOptions {
+  /**
+   * ID of the license key instance to retrieve
+   */
+  id: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"license-key">;
+}
+
+export interface GetWebhooksOptions extends PaginatedOptions {
+  /**
+   * Filter webhooks by store
+   */
+  storeId?: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"store">;
+}
+
+export interface GetWebhookOptions {
+  /**
+   * ID of the license key instance to retrieve
+   */
+  id: number;
+  /**
+   * List of record types to include
+   */
+  include?: Array<"store">;
+}
+
+
+export interface CreateWebhookOptions {
+  /**
+   * ID of the store the webhook is for
+   */
+  storeId: number;
+  /**
+   * Endpoint URL that the webhooks should be sent to
+   */
+  url: string;
+  /**
+   * List of webhook events to receive
+   * 
+   * @see https://docs.lemonsqueezy.com/help/webhooks#event-types
+   */
+  events: string[];
+  /**
+   * Signing secret (between 6 and 40 characters
+   */
+  secret: string;
+}
```

Comment:
Since we're providing a link to the documentation for the possible webhook event types we may as well type them correctly 🙂

The `WebhookEvent` can also be applied to the `UpdateWebhookOptions` types `events` property below as well.

```suggestion
type WebhookEvent =
  | "order_created"
  | "order_refunded"
  | "subscription_created"
  | "subscription_updated"
  | "subscription_cancelled"
  | "subscription_resumed"
  | "subscription_expired"
  | "subscription_paused"
  | "subscription_unpaused"
  | "subscription_payment_success"
  | "subscription_payment_failed"
  | "subscription_payment_recovered"
  | "license_key_created"
  | "license_key_updated";

export interface CreateWebhookOptions {
  /**
   * ID of the store the webhook is for
   */
  storeId: number;
  /**
   * Endpoint URL that the webhooks should be sent to
   */
  url: string;
  /**
   * List of webhook events to receive
   * 
   * @see https://docs.lemonsqueezy.com/help/webhooks#event-types
   */
  events: Array<WebhookEvent>;
  /**
   * Signing secret (between 6 and 40 characters
   */
  secret: string;
}
```